homeassistant/components/airthings_ble/sensor.py

```diff
@@ -189,4 +190,12 @@ def __init__(
     @property
     def native_value(self) -> StateType:
         """Return the value reported by the sensor."""
-        return self.coordinator.data.sensors[self.entity_description.key]
+        _LOGGER.debug(
+            "updating %s, data: %s", self.entity_description.key, self.coordinator.data
+        )
+        current_value = self.coordinator.data.sensors.get(self.entity_description.key)
+
+        if current_value is not None:
```

Comment:
You could do something like:
```python
if current_value:
  self._attr_assumed_state = False
  self._last_valid_value = current_value
else:
  self._attr_assumed_state = True
```

Reply:
I don't think a timeout is needed (it would basically be your update interval)

Reply:
Thanks for this!